common/elasticsearch/common.go

```diff
@@ -67,3 +84,23 @@ func GetESDocDelimiter() string {
 func GenerateDocID(wid, rid string) string {
 	return wid + esDocIDDelimiter + rid
 }
+
+// refer to https://github.com/olivere/elastic/blob/release-branch.v7/recipes/aws-connect-v4/main.go
+func signingClientFromStatic(credentialConfig config.AWSStaticCredential) (*http.Client, error) {
+	awsCredentials := credentials.NewStaticCredentials(
+		credentialConfig.AccessKey,
+		credentialConfig.SecretKey,
+		credentialConfig.SessionToken,
+	)
+	return esaws.NewV4SigningClient(awsCredentials, credentialConfig.Region), nil
+}
+
+func signingClientFromEnv(credentialConfig config.AWSEnvironmentCredential) (*http.Client, error) {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(credentialConfig.Region)},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return esaws.NewV4SigningClient(sess.Config.Credentials, credentialConfig.Region), nil
```

Comment:
v6 and v7 are using different esaws packages

Reply:
they do, but all the code is the same. 

Will provide a local signer to remove dependency on olivere es client